### ISSUE TITLE
Create new parameters for the economic representation in the multi-year investments

### ISSUE THREAD

**Issue by datejada** [spine-tools/SpineOpt.jl#909]
Taking as a reference the changes in the file `economic_structure` in PR #448 we need:

- [x] List the new parameters and relationships with explanations in this issue
- [x] Create the new parameters and relationships with their defaults in the SpineOpt template
- [x] Create a new case study in JSON format from the template with data (maybe future use as the tutorial)
- [x] Create a new file with the economic representation and add the function from #448 
- [x] Refactor the code as suggested in #448
- [x] Create the new tests for the economic representation
- [ ] Create migration script for renamed parameters

Notice that with this issue, we just create the parameters and calculations but we don't use them yet in the model

**Comment by manuelma** [comment]
Here it would be very useful to have a list of the parameters and their meaning.

**Comment by gnawin** [comment]
@manuelma most of the information is indeed in #476 but they can improved for sure. I can start by listing and explain things here, and later we see where the information is best placed. Note that they may be repetitive to what is there already in #476, but I agree at the moment it is important to explain more without worrying too much about being repetitive.

**Comment by manuelma** [comment]
Thank you @gnawin - I need to go through https://github.com/spine-tools/SpineOpt.jl/discussions/476 indeed.

**Comment by gnawin** [comment]
Below is a list of parameters for current design and the new design. More explanation of the parameters (including a document with more details) is in discussion #476.


# Current design

## Current parameters
`connection_investment_lifetime`, `storage_investment_lifetime`, `unit_investment_lifetime`


# New design

## Modified parameters
`connection_investment_tech_lifetime`, `storage_investment_tech_lifetime`, `unit_investment_tech_lifetime`


## New parameters

### User-defined parameters

- discount: `discount_rate`, `discount_year`

- lead time: `unit_lead_time`, `connection_lead_time`, `storage_lead_time`

- life time: `connection_investment_economic_lifetime`, `storage_investment_economic_lifetime`, `unit_investment_economic_lifetime`

### Internally calculated parameters

- annuities: `unit_conversion_to_discounted_annuities`, `connection_conversion_to_discounted_annuities`, `storage_conversion_to_discounted_annuities`
- capacity transfer factor: `unit_capacity_transfer_factor`, `connection_capacity_transfer_factor`, `storage_capacity_transfer_factor`
- salvage: `unit_salvage_fraction`, `connection_salvage_fraction`, `storage_salvage_fraction`
- tech-specific discount: `unit_tech_discount_factor`, `connection_tech_discount_factor`, `storage_tech_discount_factor`
- discount duration: `unit_discounted_duration`, `connection_discounted_duration`, `storage_discounted_duration`
- decommission discount: `unit_decommissioning_conversion_to_discounted_annuities`, `connection_decommissioning_conversion_to_discounted_annuities`, 
`storage_decommissioning_conversion_to_discounted_annuities`

**Comment by gnawin** [comment]
Hi @manuelma, I have listed the relevant items for the current design vs the new design. 

And there is a draft PR #929. The tests are yet to be added but I'd like to hear your thoughts before I continue. Please let me know what you think. Thank you very much!

**Comment by Tasqu** [comment]
I'm not especially experienced with investment optimisation, so I'm hardly the best person to comment on the formulation etc. I can comment on some aspects of the code and model structure, though. For example, in the above message, you claim:

> ## New relationships
> `model__stochastic_structure`: we currently only have default, but we need this to differentiate investment
> `model__temporal_block`: we currently only have default, but we need this to differentiate investment

As part of the new design. However, I'm a bit confused as those relationship classes already exist in SpineOpt. The `model__default_...` relationship classes are there solely for convenience, and the corresponding specific relationship classes are the more "fundamental ones". The default relationship classes are used only in cases where no specific relationships are defined.

I'm guessing you meant that you need to create new `model__investment_stochastic_structure` and `model__investment_temporal_block` relationship classes?

**Comment by DillonJ** [comment]
@manuelma correct me if I'm wrong, but the `model_...` relatioship classes were removed because they are not needed thanks to the simplification of the structure for using Benders.

The more fundamental relationships are `unit_invesment_temporal_block` which determines the temporal structure of the unit investments, which can be different for different units. model_dafault_investment_temporal_block will simply trigger creation of `unit_invesment_temporal_block` relationships where none exist.

**Comment by manuelma** [comment]
> @manuelma correct me if I'm wrong, but the `model_...` relatioship classes were removed because they are not needed thanks to the simplification of the structure for using Benders.
> 
> The more fundamental relationships are `unit_invesment_temporal_block` which determines the temporal structure of the unit investments, which can be different for different units. model_dafault_investment_temporal_block will simply trigger creation of `unit_invesment_temporal_block` relationships where none exist.

Yes, this is entirely correct.

**Comment by gnawin** [comment]
Thank you very much @DillonJ @manuelma @Tasqu. I got the exact feedback I was looking for, indeed that was some legacy code and some misunderstanding of how they work. For our purpose, using `model_dafault_investment_temporal_block` and `unit_investment_temporal_block` should be enough. I changed the design accordingly.

If you have further comments regarding the design or whatever that you think would help that leads to the merge of the PR, that would be great!
